vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/ExtendedQueryCommandCodec.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2019,2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.vertx.db2client.impl.codec;
+
+import java.util.stream.Collector;
+
+import com.ibm.db2.jcc.am.ResultSet;
+
+import io.netty.buffer.ByteBuf;
+import io.vertx.core.impl.logging.Logger;
+import io.vertx.core.impl.logging.LoggerFactory;
+import io.vertx.db2client.impl.drda.DRDAQueryRequest;
+import io.vertx.db2client.impl.drda.DRDAQueryResponse;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.Tuple;
+import io.vertx.sqlclient.impl.RowDesc;
+import io.vertx.sqlclient.impl.command.CommandResponse;
+import io.vertx.sqlclient.impl.command.ExtendedQueryCommand;
+
+class ExtendedQueryCommandCodec<R> extends ExtendedQueryCommandBaseCodec<R, ExtendedQueryCommand<R>> {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(ExtendedQueryCommandCodec.class);
+	
+    ExtendedQueryCommandCodec(ExtendedQueryCommand<R> cmd) {
+        super(cmd);
+        if (cmd.fetch() > 0) {
+            // restore the state we need for decoding fetch response
+            columnDefinitions = statement.rowDesc.columnDefinitions();
+        }
+        // @AGG always carry over column defs?
+        columnDefinitions = statement.rowDesc.columnDefinitions();
+        querySection = statement.section;
+    }
+
+    @Override
+    void encode(DB2Encoder encoder) {
+        super.encode(encoder);
+        if (LOG.isDebugEnabled())        
+        	LOG.debug("Extended query encode: statement=" + statement);
+        
+        ByteBuf packet = allocateBuffer();
+        DRDAQueryRequest openQuery = new DRDAQueryRequest(packet);
+        String dbName = encoder.socketConnection.database();
+        int fetchSize = 0; // TODO @AGG get fetch size from config
+        Tuple params = cmd.params();
+        Object[] inputs = new Object[params.size()];
+        for (int i = 0; i < params.size(); i++)
+            inputs[i] = params.getValue(i);
+        
+        if (DRDAQueryRequest.isQuery(cmd.sql())) {
```

Comment:
so the protocol used depends on the interpretation of the SQL query ?

Reply:
The protocol used is always DRDA, but yes we need to send different commands over the wire for an update vs. query statement. I believe that we technically could use the same commands for both, but if we flow the more specific commands then the DB can make some optimizations on its end